internal/datacoord/server.go

```diff
@@ -994,7 +1008,16 @@ func (s *Server) reCollectSegmentStats(ctx context.Context) {
 	nodes := s.sessionManager.getLiveNodeIDs()
 	log.Info("re-collecting segment stats from DataNodes",
 		zap.Int64s("DataNode IDs", nodes))
-	for _, node := range nodes {
-		s.cluster.ReCollectSegmentStats(ctx, node)
+
+	reColllectFunc := func() error {
+		err := s.cluster.ReCollectSegmentStats(ctx)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if err := retry.Do(ctx, reColllectFunc, retry.Attempts(20)); err != nil {
```

Comment:
`Attempts(20)` will cost about 50 seconds. May caused unwanted panic when cluster restarts and datanodes' sessions remains in etcd (worst case 60 seconds)

Reply:
set the max retry time to 5s to avoid the corner case.
We will change etcd lease timeout to 20s